joblib/parallel.py

```diff
@@ -129,12 +190,238 @@ def get_active_backend(prefer=None, require=None, verbose=0):
+        - 'multiprocessing': legacy single-host, process-based parallelism.
+
+        'loky' is recommended to run functions that manipulate Python objects.
+        'threading' is a low-overhead alternative that is most efficient for
+        functions that release the Global Interpreter Lock: e.g. I/O-bound
+        code or CPU-bound code in a few calls to native code that explicitly
+        releases the GIL. Note that on some rare systems (such as pyiodine),
+        multiprocessing and loky may not be available, in which case joblib
+        defaults to threading.
+
+        In addition, if the `dask` and `distributed` Python packages are
+        installed, it is possible to use the 'dask' backend for better
+        scheduling of nested parallel calls without over-subscription and
+        potentially distribute parallel calls over a networked cluster of
+        several hosts.
+
+        It is also possible to use the distributed 'ray' backend for
+        distributing the workload to a cluster of nodes. To use the 'ray'
+        joblib backend add the following lines::
+
+        >>> from ray.util.joblib import register_ray  # doctest: +SKIP
+        >>> register_ray()  # doctest: +SKIP
+        >>> with parallel_backend("ray"):  # doctest: +SKIP
+        ...     print(Parallel()(delayed(neg)(i + 1) for i in range(5)))
+        [-1, -2, -3, -4, -5]
+
+        Alternatively the backend can be passed directly as an instance.
+
+    n_jobs : int, default=None
+        The maximum number of concurrently running jobs, such as the number
+        of Python worker processes when backend="multiprocessing"
+        or the size of the thread-pool when backend="threading".
+        If -1 all CPUs are used. If 1 is given, no parallel computing code
+        is used at all, which is useful for debugging. For n_jobs below -1,
+        (n_cpus + 1 + n_jobs) are used. Thus for n_jobs = -2, all
+        CPUs but one are used.
+        None is a marker for 'unset' that will be interpreted as n_jobs=1.
+
+    verbose : int, default=0
+        The verbosity level: if non zero, progress messages are
+        printed. Above 50, the output is sent to stdout.
+        The frequency of the messages increases with the verbosity level.
+        If it more than 10, all iterations are reported.
+
+    temp_folder : str, default=None
+        Folder to be used by the pool for memmapping large arrays
+        for sharing memory with worker processes. If None, this will try in
+        order:
+
+        - a folder pointed by the JOBLIB_TEMP_FOLDER environment
+            variable,
+        - /dev/shm if the folder exists and is writable: this is a
+            RAM disk filesystem available by default on modern Linux
+            distributions,
+        - the default system temporary folder that can be
+            overridden with TMP, TMPDIR or TEMP environment
+            variables, typically /tmp under Unix operating systems.
```

Comment:
```suggestion
        Folder to be used by the pool for memmapping large arrays
        for sharing memory with worker processes. If None, this will try in
        order:

        - a folder pointed by the `JOBLIB_TEMP_FOLDER` environment
          variable,
        - `/dev/shm` if the folder exists and is writable: this is a
          RAM disk filesystem available by default on modern Linux
          distributions,
        - the default system temporary folder that can be
          overridden with `TMP`, `TMPDIR` or `TEMP` environment
          variables, typically `/tmp` under Unix operating systems.
```